packages/moda-sdk/src/profile/createProfile.ts

```diff
@@ -0,0 +1,33 @@
+import { Hash } from 'viem';
+import { ProfileABI } from './abi/ProfileABI';
+import { profileAddress } from './constants/profileAddress';
+import { CreateProfileArgs } from './types';
+
+export async function createProfile(
+  createProfileArgs: CreateProfileArgs,
+): Promise<Hash> {
+  const { config, metadataUri } = createProfileArgs;
+  const { publicClient, walletClient } = config;
+
+  if (publicClient === undefined) {
+    throw new Error('Public client is undefined');
+  }
+
+  if (walletClient === undefined) {
+    throw new Error('Wallet client is undefined');
```

Comment:
If you implemented custom error strings as an enum then you would not throw here, but instead do:

```ts
return CreateProfileError.ClientUndefined;
```